File: jpa/src/test/java/me/jiangcai/common/jpa/entity/Foo_.java
package me.jiangcai.common.jpa.entity;

import javax.annotation.Generated;
import javax.persistence.metamodel.SingularAttribute;
import javax.persistence.metamodel.StaticMetamodel;
import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * @author CJ
 */
@Generated(value = "org.hibernate.jpamodelgen.JPAMetaModelEntityProcessor")
@StaticMetamodel(Foo.class)
public class Foo_ {

    public static volatile SingularAttribute<Foo, String> id;
    public static volatile SingularAttribute<Foo, BigDecimal> value;
    public static volatile SingularAttribute<Foo, LocalDateTime> created;
}

File: jpa/src/test/kotlin/me/jiangcai/common/jpa/entity/Foo.kt
package me.jiangcai.common.jpa.entity

import me.jiangcai.common.jpa.type.JSONStoring
import java.math.BigDecimal
import java.time.LocalDateTime
import javax.persistence.*

/**
 * @author CJ
 */
@Entity
class Foo {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null
    @Column(precision = 12, scale = 10)
    var value: BigDecimal? = null
    @Column(columnDefinition = "datetime")
    var created: LocalDateTime = LocalDateTime.now()
    @Column(columnDefinition = "text")
//    @Convert(converter = JSONStoringConverter::class)
    var goodThing: JSONStoring? = null
//    @Convert(converter = GoodThingConverter::class)
//    var goodThing: GoodThing? = null
}